### DISCUSSION TITLE
Display and floppies

### DISCUSSION THREAD

**Discussion by nicolasbauw** [dirkwhoffmann/vAmiga#691] [Ideas]
Hi,
At first I would like to say that I love the emulator. I currently use the 1.02 version (via homebrew).
I'd like to share a few things that I noticed and some proposals:

- the dot mask and scanlines parameters are really great, and make the experience a lot more immersive, to me it is a very important point. Everyone can set it to its taste. What about a "Revert to..." preset that mimics the rendering of a 1084 monitor which was very common and had a great image ? Note : I think that in "embedded" mode, the scanlines are too thick / too visible.
- when I write a file to an adf, i can see the file until I quit the emulator. When I have a closer look, I can see that the file is not written to the adf file on the host disk (probably it's written in a sort of internal working copy of the adf ?) that's a problem for games saving progression to a save disk.
- I find that the emulated sound for floppy head seeking is not very realistic. For example, I've played and still play a lot to Lotus 2 (both on real Amiga and emulation) which has a very noticeable drive seek sound when loading, and it sounds completely different in the emulator.
- When I set my display at 50 Hz, I would expect to obtain a very smooth scrolling (scrolltexts in demos...) like in winUAE, and like original Amigas, but that's not the case. In fact, it's even worse than at 60 Hz ! For example in the Enigma demo, which has a very fast scrolltext, it's very noticeable (and nearly unreadable). I did not see something like a "vsync" option. A nice option would also be the possibility to switch to 50 Hz when in fullscreen.

Cheers,
Nicolas

**Comment by dirkwhoffmann** [comment]
Hi Nicolas,

thank you very much for your detailed feedback. You are right about the 50 Hz mode. There is no such thing as an VSYNC option yet. This is something that has to be implemented in the near future. 

A preset for the 1084 would be nice, but somebody with a real 1084 would have to determine the parameters or implement a custom dot mask. I don't have such a monitor and thus won't be able to do it myself. 

ADFs don't support "write through". This is by design. I wanted to make sure that ADFs can't get modified or damaged by any software running inside the emulator. However, disks can easily be exported (drive menu options). 

I'm open to replace the drive sounds. If somebody can provide more realistic ones, please do! Your comment might be especially true for hard drive sounds. Because I didn't find realistic ones, I ended up using some freely available click sound. They sound OK, but aren't very realistic I guess.

**Comment by nicolasbauw** [comment]
Hi !

I actually own a 1084S, but for the moment I cannot use it. That said, I've spent really a lot of time with this monitor, and from my memory (a few years ago) and after some hours of tuning, I've set up for myself this video configuration:

<img width="646" alt="image" src="https://user-images.githubusercontent.com/37193744/169713856-cff568fb-2263-4002-9d5e-37247db37bae.png">

It could be used as a starting point, maybe other people could also tune these settings based on their own experience and share their settings.

**Comment by nicolasbauw** [comment]
This morning I saw this CRT shader, and found it very realistic:
![image](https://user-images.githubusercontent.com/37193744/208451000-fec5b4c3-f6c6-44e2-9cb0-dd5b131d059b.png)

**Comment by dirkwhoffmann** [comment]
> This morning I saw this CRT shader, and found it very realistic:

Looks great! Is it open source?

**Comment by nicolasbauw** [comment]
It's GPL-3, here is the repo:
https://github.com/mattgodbolt/jsbeeb/tree/new_canvas_glb

**Comment by dirkwhoffmann** [comment]
I didn't browse through all the code, but a good starting point seems to be in `scene/beep.js`: 

```js
    async load() {
          document.getElementById('loading-status').innerHTML="Loading CRT shader";
            const [maskTexture, screenPrologFragment, screenEmissiveFragment, screenEpilogFragment, model] = await Promise.all(
                [
                    loadMaskTexture(),
                    loadShaderSource('scene/screen_prolog.glsl'),
                    loadShaderSource('scene/screen_emissive.glsl'),
                    loadShaderSource('scene/screen_epilog.glsl'),
                    loadModel()
                ]);
            this.screenMaterial = this.makeScreenMaterial(maskTexture, screenPrologFragment, screenEmissiveFragment, screenEpilogFragment);
            this.model = this.prepareScene(model);
        }
```

Maybe @mithrendal is interested to integrate the shader in the web version. Since it's JavaScript, it might be doable without too much effort.
